keps/sig-apps/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,453 @@
+    1. Create statefulset in previous version and upgrade to the version 
+       supporting this feature. The PVCs should remain intact.
+    2. Downgrade to earlier version and check the PVCs with Retain
+       remain intact and the others with set policies before upgrade 
+       gets deleted based on if the references were already set.
+1. Feature disablement/enable test for alpha feature flag `statefulset-autodelete-pvcs`.
+
+
+### Graduation Criteria
+
+#### Alpha release
+- Complete adding the items in the 'Changes required' section.
+- Add unit, functional, upgrade and downgrade tests to automated k8s test.
+
+### Upgrade / Downgrade Strategy
+
+This features adds a new field to the StatefulSet. The default value for the new field
+maintains the existing behavior of StatefulSets.
+
+On a downgrade, the PersistentVolumeClaimReclaimPolicy field will be removed from any
+StatefulSets. If a scale-down or StatefulSet deletion is in process when the downgrade
+happens, any existing ownerRefs on PVCs will not be removed so that in most cases when the
+scale-down completes, unused PVCs will be deleted. However, there may be edge cases causing
+only some of the unused PVCs to be deleted. As unused PVCs remaining after a scaledown is
+the expected behavior of the downgraded clusters no further effort will be made to remove
+them.
+
+### Version Skew Strategy
+There is only kubecontroller manager changes involved, hence not applicable for
+version skew involving other components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: StatefulSetAutoDeletePVC
+    - Components depending on the feature gate: kube-controller-manager
+  
+* **Does enabling the feature change any default behavior?**
+  No. What happens during StatefulSet deletion differs from current behaviour
+  only when the user explicitly specifies the
+  `PersistentVolumeClaimDeletePolicy`.  Hence no change in any user visible
+  behaviour change by default.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?**
+  Yes. Disabling the feature gate will cause the new field to be ignored. If the feature
+  gate is re-enabled, the new behavior will start working.
+  
+  There are new corner cases here. For example, if a StatefulSet deletion is in process
+  during a StatefulSet deletion, the appropriate ownerRefs will not have been added and
+  PVCs may not be deleted. The exact behavior will be discovered during feature
+  testing. In any case the mitigation will be to manually delete any PVCs.
+  
+* **What happens if we reenable the feature if it was previously rolled back?**  
+  The reconcile loop which removes references on disablement will not come into
+  action. Since the StatefulSet field would persist through the disablment we will have to
```

Comment:
I think I don't understand the first sentence - can you please clarify?

Reply:
Clarified.